contracts/SnappAuction.sol

```diff
@@ -0,0 +1,81 @@
+pragma solidity ^0.5.0;
+
+import "./SnappBase.sol";
+
+
+contract SnappAuction is SnappBase {
+  
+    uint16 public constant AUCTION_BATCH_SIZE = 1000;
+
+    uint public auctionIndex;
+    mapping (uint => PendingFlux) public auctions;
+
+    event SellOrder(
+        uint auctionId, 
+        uint16 slotIndex,
+        uint16 accountId, 
+        uint8 buyToken, 
+        uint8 sellToken, 
+        uint128 buyAmount,
+        uint128 sellAmount
+    );
+
+    constructor () public {
+        auctions[auctionIndex].creationBlock = block.number;
+    }
+
+    function placeSellOrder(
+        uint8 buyToken,
+        uint8 sellToken,
+        uint128 minBuyAmount,
+        uint128 maxSellAmount
+    ) public onlyRegistered() {
+        require(buyToken != sellToken, "Buy and Sell tokens must differ!");
+
+        // Must have 0 < tokenId < MAX_TOKENS anyway, so might as well ensure registered.
+        require(tokenIdToAddressMap[buyToken] != address(0), "Buy token is not registered");
+        require(tokenIdToAddressMap[sellToken] != address(0), "Sell token is not registered");
+
+        // Restrict buy and sell amount to occupy less than 100 bits.
+        require(minBuyAmount < 0x10000000000000000000000000, "Buy amount too large!");
+        require(maxSellAmount < 0x10000000000000000000000000, "Sell amount too large!");
+
+        if (
+            auctions[auctionIndex].size == AUCTION_BATCH_SIZE || 
+            block.number > auctions[auctionIndex].creationBlock + 20
+        ) {
+            auctionIndex++;
+            auctions[auctionIndex] = PendingFlux({
```

Comment:
I'm not a big fan of reusing PendingFlux here - since a flux is a deposit or withdraw. If the structure has the same data, maybe we can rename PendingFlux to PendingBatch?

Reply:
I am in favour of `PendingBatch`